### DISCUSSION TITLE
How best to detect MySQL injection on database side?

### DISCUSSION THREAD

**Discussion by JJGadgets** [Security-Onion-Solutions/securityonion#8116] [Unsupported Versions]
I am running SecurityOnion standalone as a PoC for an evaluation project, and I would like to simulate running and then detecting SQL injection MySQL queries. 

I am running SecurityOnion 2.3.110, CentOS 7 for the MySQL server, and MySQL server v5.5.22 (due to CVEs available, this is a PoC not production)

I am thinking that I should log all MySQL queries, either using MySQL's general_log on the host running MySQL server, or using NIDS. Then, rules on the SO side will analyze all queries and trigger alerts for those that resemble SQL injection.

I have tried the following, all to no avail:

- ingesting MySQL general_log to SO using Wazuh agent's <localfile>mysql_log</localfile> option pointed to /var/log/mysql/mysql.log (my general_log file)
- the same as above, but using Filebeat installed on MySQL server host, using its MySQL module, ingesting to SecurityOnion's Logstash
- trying to log MySQL queries using a custom Suricata rule (`alert ip any any -> any 3306 (msg: "MySQL query"; sid:2022202222;)`). I have verified that my NIDS is working properly, since the Suricata rule from ET Open, "ET SCAN Suspicious inbound to mySQL port 3306", is being logged.

All 3 methods I've tried have logged **no** SQL queries, let alone SQL *injection* queries. Anyone have any ideas or suggestions as to my approach and how to continue from here?

**Comment by dougburks** [comment]
If your custom Suricata rule is not firing, have you double-checked that you followed all steps here?
https://docs.securityonion.net/en/2.3/local-rules.html#nids
